### PR TITLE
Types refactoring

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

wrapping commands with Promises increased complexity too much so I decided to generate two separate files, one for async and another for sync versions.

- avoid wrapping commands with Promise 
- generate separate files for sync and async versions so there should not be any hidden/tricky/unobvious issues
- add generated types files to the repo to detect changes in webdriver, webdriverio and @wdio/sync packages automatically
- ~dropped WebdriverIOAsync, so it has to be replaced with WebdriverIO~ let's do it in v6 https://github.com/webdriverio/webdriverio/issues/4720
- added js docs https://github.com/webdriverio/webdriverio/issues/4674

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mgrybyk** [comment]
Do you think it's a serious change to remove WebdriverIOAsync and use only WebdriverIO?

ping @CrispusDH as far as he is one of the typescript + async mode users

**Comment by mgrybyk** [comment]
webdriverio command
image

webdriverio command from specific commands
image

protocol command
image

**Comment by mgrybyk** [comment]
@CrispusDH yes. As a typescript + async user, is it difficult to replace WebdriverIOAsync with WebdriverIO in your project?
I want to understand can we do it now or not

**Comment by christian-bromann** [comment]
Happy to merge this one after getting feedback from @CrispusDH

**Comment by mgrybyk** [comment]
@christian-bromann what is you opinion that the PR is breaking? 
Sorry to say there is no way to keep both old and new behavior. 
We may postpone it till v6, of course I'd like to have it now, hehe

**Comment by christian-bromann** [comment]
Let's understand first if it would break @CrispusDH tests. Do you see a way to make this an opt-in thing?

**Comment by mgrybyk** [comment]
I have tried in multiple ways but can't make it optin. I haven't tried on more thing I have on my mind which is to leave async typings as is and replace them with new way in v6

**Comment by erwinheitzman** [comment]
Will have a look at this tomorrow @mgrybyk

**Comment by CrispusDH** [comment]
> @CrispusDH yes. As a typescript + async user, is it difficult to replace WebdriverIOAsync with WebdriverIO in your project?
> I want to understand can we do it now or not

I can do it very easily. Using WebStorm I will replace `WebdriverIOAsync` with `WebdriverIO`. I think VSCode should have some similar functionality for refactoring.

**Comment by mgrybyk** [comment]
Same here, but it's still breaking  :(

**Comment by christian-bromann** [comment]
Let's hold this for v6 release and plan this release earlier.

**Comment by mgrybyk** [comment]
now it's not be a breaking change, created https://github.com/webdriverio/webdriverio/issues/4720 to do it for v6

File: scripts/templates/webdriverio.tpl.d.ts [hunk 1]
```diff
@@ -176,17 +180,25 @@ declare namespace WebdriverIO {
...
+         * overwrite `browser` or `element` command
+         */
         overwriteCommand(
             name: string,
             func: (origCommand: Function, ...args: any[]) => any,
             attachToElement?: boolean
         ): void;
-        options: RemoteOptions;
         // ... browser commands ...
     }
```
Comment (baruchvlz) [inline scripts/templates/webdriverio.tpl.d.ts]: We should add the type definitions for `addLocatorStrategy` missed these in my PR.

Comment (mgrybyk) [reply scripts/templates/webdriverio.tpl.d.ts]: Will you be able to raise a PR for that?

**Pull request merged**